### ISSUE TITLE
instances metric does not work

### ISSUE THREAD

**Issue by sarapapi** [facebookresearch/SimulEval#35]
Hi @xutaima,
in using the new version of SimulEval, I noticed that the metrics per instance are not shown in the instances.log output. An example  from your quick start:
```
{"index": 0, "prediction": "M N P V M S S O C M T", "delays": [3, 4, 5, 6, 7, 8, 9, 10, 11, 11, 11], "elapsed": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], "prediction_length": 11, "reference": "Z U S N B Y X Q L O T\n", "source": "Z U S N B Y X Q L O T", "source_length": 11, "metric": {}}
{"index": 1, "prediction": "O I R B D G Y V Z X M E Q M W M", "delays": [3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 16, 16], "elapsed": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], "prediction_length": 16, "reference": "M A J F P G O Y V R H M Z O T A\n", "source": "M A J F P G O Y V R H M Z O T A", "source_length": 16, "metric": {}}
{"index": 2, "prediction": "Z N H E X D Y T X W C B", "delays": [3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 12, 12], "elapsed": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], "prediction_length": 12, "reference": "M O W A O I D H H B O F\n", "source": "M O W A O I D H H B O F", "source_length": 12, "metric": {}}
{"index": 3, "prediction": "Y M Q D D N J G B M", "delays": [3, 4, 5, 6, 7, 8, 9, 10, 10, 10], "elapsed": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0], "prediction_length": 10, "reference": "N Q N I P C O H A A\n", "source": "N Q N I P C O H A A", "source_length": 10, "metric": {}}
{"index": 4, "prediction": "J N V G H D X S B C W B", "delays": [3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 12, 12], "elapsed": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], "prediction_length": 12, "reference": "G B O J H P W C I A L V\n", "source": "G B O J H P W C I A L V", "source_length": 12, "metric": {}}
{"index": 5, "prediction": "I J X Z S R M Q N S E D F N", "delays": [3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 14, 14], "elapsed": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], "prediction_length": 14, "reference": "P T Z D E E N T B Y G Z R K\n", "source": "P T Z D E E N T B Y G Z R K", "source_length": 14, "metric": {}}
{"index": 6, "prediction": "Y X D A R B Z Z Z V L W W D D A V K Y C", "delays": [3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 20, 20], "elapsed": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], "prediction_length": 20, "reference": "F S H U K R W K S B R K M B B Q F C O U\n", "source": "F S H U K R W K S B R K M B B Q F C O U", "source_length": 20, "metric": {}}
{"index": 7, "prediction": "A H J J I J W T R C F H B", "delays": [3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 13, 13], "elapsed": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], "prediction_length": 13, "reference": "M H O L W Z G J Y X J B I\n", "source": "M H O L W Z G J Y X J B I", "source_length": 13, "metric": {}}
{"index": 8, "prediction": "U F Y E H E G C E V F W", "delays": [3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 12, 12], "elapsed": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], "prediction_length": 12, "reference": "A V B F E S F E W Q C S\n", "source": "A V B F E S F E W Q C S", "source_length": 12, "metric": {}}
{"index": 9, "prediction": "S Y J B A T K W I T D I Y", "delays": [3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 13, 13], "elapsed": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], "prediction_length": 13, "reference": "F N O I E Z B R S C V N S", "source": "F N O I E Z B R S C V N S", "source_length": 13, "metric": {}}
```
And even if I manually add quality/latency metrics to the script, the metrics are still not present in the output while the global scores are instead computed and saved in scores.tsv: 
```
BLEU	LAAL	AL	AP	DAL	ATD
1.353	3.0	3.0	0.676	3.0	3.0
```
Thanks

**Comment by xutaima** [comment]
Hi @sarapapi, sorry for the late response. The current way to compute the metric is done by the end of the whole evaluation instead of each individual instance. I will make a PR to save the individual metrics as well.

**Comment by xutaima** [comment]
Hi @sarapapi, the metrics of instances not saved in another file implemented in #40. Feel free to let me know if you have any questions.
